servo/connectors/kubernetes.py

```diff
@@ -2003,6 +2016,578 @@ async def get_restart_count(self) -> int:
 
         return count
 
+# Workarounds to allow use of api_client.deserialize() public method instead of private api_client._ApiClient__deserialize
+# TODO: is this workaround worth it just to avoid using the private method?
+# fix for https://github.com/kubernetes-client/python/issues/977#issuecomment-594045477
+def default_kubernetes_json_serializer(o: Any) -> Any:
+    if isinstance(o, (datetime.datetime, datetime.date)):
+        return o.isoformat()
+    raise TypeError(f'Object of type {o.__class__.__name__} '
+                        f'is not JSON serializable')
+
+# https://github.com/kubernetes-client/python/issues/977#issuecomment-592030030
+class FakeKubeResponse:
+    """Mocks the RESTResponse object as a workaround for kubernetes python api_client deserialization"""
+    def __init__(self, obj):
+        self.data = json.dumps(obj, default=default_kubernetes_json_serializer)
+
+# Use alias generator so that lower camel case can be parsed to snake case properties to match k8s python client behaviour
+def to_lower_camel(string: str) -> str:
+    split = string.split('_')
+    return split[0] + ''.join(word.capitalize() for word in split[1:])
```

Comment:
Prolly should live up in the types module and needs some unit tests. The naming seems unintuitive as it assumes you have a string delimited by an underscore. The comment is perplexing as well. How can you have a `lower camel case` string? Isn't camel-case defined entirely by having a mix of upper and lower case characters delimiting the boundary of the components?

Reply:
1. WRT the method name, the input for this method isn't the field name of the kubernetes object, rather its the fields of the pydantic model. Eg. deletion_grace_period_seconds -> deletionGracePeriodSeconds
2. Lower camel case seemed to be the industry accepted term for having the first word be lower case followed by standard camel case but I will make an update to clarify this with the more formal naming of "dromedary case"